### PR TITLE
Authentication and build UI

### PR DESCRIPTION
**Pull request by katiefenn**
- [x] Add test for Login with GitHub button
- [x] Add test for progress icons
- [x] Replace emoji progress icons with something more appropriate
- [x] Add test for webhook reducer
- [x] Add test for webhook actions
- [x] Fix linting errors
- [x] Rebase changes from master
- [x] Merge PR #23

### CODE REVIEW

File: environments/staging.env [hunk 1]
```diff
@@ -7,3 +7,10 @@ UBUNTU_SSO_URL=https://login.ubuntu.com
 OPENID_VERIFY_URL=https://build.staging.snapcraft.io/login/verify
 OPENID_TEAMS=["ubuntuone-hackers"]
 LP_API_URL=https://api.launchpad.net
+WEBHOOK_ENDPOINT=https://build.staging.snapcraft.io/webhook/notify
+GITHUB_API_ENDPOINT=https://api.github.com
+GITHUB_AUTH_LOGIN_URL=https://github.com/login/oauth/authorize
+GITHUB_AUTH_VERIFY_URL=https://github.com/login/oauth/access_token
+GITHUB_AUTH_REDIRECT_URL=https://build.staging.snapcraft.io/auth/verify
+GITHUB_AUTH_CLIENT_ID=901f5e1b1efa766d0ff1
+GITHUB_AUTH_CLIENT_SECRET=db582f35e8790f94f831f57be58ca24690e20052
```
Comment (cjwatson) [inline environments/staging.env]: This is probably a leftover of something; staging creds shouldn't be committed.

File: src/server/handlers/github.js [hunk 2]
```diff
@@ -86,3 +72,23 @@ export const newIntegration = (req, res) => {
     return res.status(201).send(RESPONSE_CREATED);
   });
 };
+
+const getRequest = (account, repo, token) => {
+  return {
+    headers: {
+      'User-Agent': 'SnapCraftBuild',
```
Comment (cjwatson) [inline src/server/handlers/github.js]: `User-Agent` is handled by `requestGitHub`.

File: src/common/actions/webhook.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return fetch(`${getBaseUrl()}/api/github/webhook`, settings)
+      .then((response) => {
+        return response.text();
+      })
+      .then((body) => {
+        let result = JSON.parse(body);
+        if (result.status && result.payload) {
+          if (result.status == 'success' || result.payload.code == 'github-already-created') {
+            // Treat pre-existing builds like a new build
+            dispatch(createSnap(repositoryString));
```
Comment (katiefenn) [inline src/common/actions/webhook.js]: Don't know what to do when createSnap fails. I guess eventually we'll have to roll back the webhook creation.

Comment (bartaz) [inline src/common/actions/webhook.js]: Maybe webhook creation should be the last step? After snap is successfully created?

**Pull request merged**